runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -8039,140 +8156,188 @@ static inline
 
-  case FI_BOR:
-    if (ofiType == FI_INT32) {
-      CPU_INT_ARITH_AMO(or, int_least32_t, i32);
-    } else if (ofiType == FI_UINT32) {
-      CPU_INT_ARITH_AMO(or, uint_least32_t, u32);
-    } else if (ofiType == FI_INT64) {
-      CPU_INT_ARITH_AMO(or, int_least64_t, i64);
-    } else if (ofiType == FI_UINT64) {
-      CPU_INT_ARITH_AMO(or, uint_least64_t, u64);
-    } else {
-      INTERNAL_ERROR_V("doCpuAMO(): unsupported ofiOp %d, ofiType %d",
-                       ofiOp, ofiType);
-    }
-    break;
+    case FI_BOR:
+      switch(ofiType){
+        INTEGER_CASES(or);
+        default: UNSUPPORTED_ERROR();
+      }
+      break;
 
-  case FI_BXOR:
-    if (ofiType == FI_INT32) {
-      CPU_INT_ARITH_AMO(xor, int_least32_t, i32);
-    } else if (ofiType == FI_UINT32) {
-      CPU_INT_ARITH_AMO(xor, uint_least32_t, u32);
-    } else if (ofiType == FI_INT64) {
-      CPU_INT_ARITH_AMO(xor, int_least64_t, i64);
-    } else if (ofiType == FI_UINT64) {
-      CPU_INT_ARITH_AMO(xor, uint_least64_t, u64);
-    } else {
-      INTERNAL_ERROR_V("doCpuAMO(): unsupported ofiOp %d, ofiType %d",
-                       ofiOp, ofiType);
-    }
-    break;
+    case FI_BXOR:
+      switch(ofiType){
+        INTEGER_CASES(xor);
+        default: UNSUPPORTED_ERROR();
+      }
+      break;
 
-  case FI_SUM:
-    if (ofiType == FI_INT32) {
-      CPU_INT_ARITH_AMO(add, int_least32_t, i32);
-    } else if (ofiType == FI_UINT32) {
-      CPU_INT_ARITH_AMO(add, uint_least32_t, u32);
-    } else if (ofiType == FI_INT64) {
-      CPU_INT_ARITH_AMO(add, int_least64_t, i64);
-    } else if (ofiType == FI_UINT64) {
-      CPU_INT_ARITH_AMO(add, uint_least64_t, u64);
-    } else if (ofiType == FI_FLOAT) {
-      CPU_INT_ARITH_AMO(add, _real32, r32);
-    } else if (ofiType == FI_DOUBLE) {
-      CPU_INT_ARITH_AMO(add, _real64, r64);
-    } else {
-      INTERNAL_ERROR_V("doCpuAMO(): unsupported ofiOp %d, ofiType %d",
-                       ofiOp, ofiType);
-    }
-    break;
+    case FI_SUM:
+      switch(ofiType) {
+        INTEGER_CASES(add);
+        REAL_CASES(add);
+        default: UNSUPPORTED_ERROR();
+      }
+      break;
 
+#define CPU_MIN_MAX_AMO(_o, _t, _m)                               \
+  do {                                                            \
+    _t _val = atomic_load_##_t(obj);                              \
+    while(myOpnd->_m _o _val) {                                   \
+      if (atomic_compare_exchange_weak_##_t(obj,                  \
+                                           &_val, myOpnd->_m)) {  \
+        if (result != NULL) {                                     \
+          *(_t*) result = _val;                                   \
+        }                                                         \
+        break;                                                    \
+      }                                                           \
+    }                                                             \
+  } while(0);
```

Comment:
nit, remove the semicolon after `while(0)`

Reply:
Thanks for catching that.